rest/route/build_task.go

```diff
@@ -105,17 +106,20 @@ func (tbh *tasksByBuildHandler) Run(ctx context.Context) gimlet.Responder {
 
 	tasks = tasks[:lastIndex]
 	for i := range tasks {
+		taskCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
```

Comment:
It seems like this doesn't make a new context, it's just wrapping the existing context in another child context, no? This would just apply an additional timeout, on top of whatever cancellation/timeouts are already embedded in the parent request context.

You would have to make a brand new `context.WithTimeout(context.Background(), 10 * time.Second)` if you want to ignore the parent context.

Reply:
🤦